snuba/request/schema.py

```diff
@@ -112,7 +114,10 @@ def validate(self, value, dataset: Dataset, referrer: str) -> Request:
                 if key in value
             }
 
-        query = parse_query(query_body, dataset)
+        if "query" in query_body:
+            query = parse_snql_query(query_body["query"], dataset)
+        else:
+            query = parse_query(query_body, dataset)
```

Comment:
I think this was supposed to be in the PR that adds the schema for the SnQL query to the request schema. At this point `query_body["query"]` should never be populated.